lifecycle-operator/webhooks/pod_mutator/handlers/objectmeta.go

```diff
@@ -94,24 +94,44 @@ func initEmptyAnnotations(meta *metav1.ObjectMeta, size int) {
 	}
 }
 
-func calculateVersion(pod *corev1.Pod) string {
+func getImageVersion(image string) (string, error) {
+	splitImage := strings.Split(image, ":")
+	lenImg := len(splitImage) - 1
+	if lenImg >= 1 && splitImage[lenImg] != "" && splitImage[lenImg] != "latest" {
+		return splitImage[lenImg], nil
+	}
+	return "", fmt.Errorf("Invalid image version")
+}
+
+func calculateVersion(pod *corev1.Pod, containerName string) (string, error) {
 	if len(pod.Spec.Containers) == 1 {
-		image := strings.Split(pod.Spec.Containers[0].Image, ":")
-		lenImg := len(image) - 1
-		if lenImg >= 1 && image[lenImg] != "" && image[lenImg] != "latest" {
-			return image[lenImg]
+		if containerName != "" && pod.Spec.Containers[0].Name != containerName {
```

Comment:
can we introduce a test to cover this error case ?

Reply:
[added test](https://github.com/keptn/lifecycle-toolkit/pull/2471/commits/f72ca17268768005480a58ba9ae437f533ba5946)